pkg/operator/upgradebackupcontroller/upgradebackupcontroller.go

```diff
@@ -195,11 +196,21 @@ func (c *UpgradeBackupController) ensureRecentBackup(ctx context.Context, cluste
 	case backupPod.Status.Phase == corev1.PodSucceeded:
 		backupCondition := configv1helpers.FindStatusCondition(clusterOperatorStatus.Conditions, backupConditionType)
 		if backupCondition.Reason != backupSuccess {
+
+			// Read the backup directory name from the backup pod's env
+			// TODO(hasbro17): backup pod has already succeeded but should we error if env unset or empty? unlikely it could succeed without the env.
```

Comment:
Agreed it should never probably happen but not sure if we should error out. Status is best-effort basis, rather have something in the status at least the type and reason, incomplete msg is better than erroring out here. WDYT?